lib/features/push_notification/presentation/controller/web_socket_controller.dart

```diff
@@ -239,4 +251,19 @@ class WebSocketController extends PushBaseController {
       logError('WebSocketController::_handleStateChange:Exception = $e');
     }
   }
+
+  void _monitorNetwork() {
+    _networkConnectionController = getBinding<NetworkConnectionController>();
+    _connectivitySubscription = _networkConnectionController
+      ?.connectivity
+      .onConnectivityChanged.listen((status) {
+        if (status == ConnectivityResult.none) {
+          log('WebSocketController::_monitorNetwork:No network connection');
+          _cleanUpWebSocketResources();
+        } else {
+          log('WebSocketController::_monitorNetwork:Network connection restore');
+          _connectWebSocket();
```

Comment:
Please verify memory leak after we turn off/on network n times

Reply:
No memory leak when continuously turning on/off the network.


https://github.com/user-attachments/assets/34dc9781-3d7c-4c81-ba8e-7f4af545b45f